hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/VersionedObjectDataOutput.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl;
+
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.VersionAware;
+import com.hazelcast.version.Version;
+
+import java.io.OutputStream;
+
+/**
+ * Base class for ObjectDataOutput that is VersionAware and allows mutating the version
+ */
+abstract class VersionedObjectDataOutput extends OutputStream implements ObjectDataOutput, VersionAware {
+
+    protected Version version = Version.UNKNOWN;
```

Comment:
what's the semantic of this? under what circumstances it can return `UKNOWN` version?